### ISSUE TITLE
Cannot build doxygen as a subproject

### ISSUE THREAD

**Issue by madebr** [doxygen/doxygen#8035]
**Describe the bug**

For building doxygen, I use the cmake build scripts.
For configuring the compiler options, I add cmake as a subproject.

Currently, doxygen does not allow this because it expects to be built as the main project.
The cmake configure step fails.

Also, recent versions of xapian have a `xapian-config.cmake` script for including it.
Doxygen currently ships with a `FindXapian.cmake` script.

This problem was encountered while creating a conan package at https://github.com/conan-io/conan-center-index/pull/2912.
(conan is a C/C++ package manager)

Thanks!

**Expected behavior**
cmake configure succeeds and we can build.

**To Reproduce**

```sh
cd /tmp
mkdir superproject
cd superproject
cat >CMakeLists.txt <<EOF
cmake_minimum_required(VERSION 3.1)
project(superproject)
add_subdirectory(doxygen)
EOF
git clone https://github.com/doxygen/doxygen.git
mkdir build
cd build
cmake ..
```

**Version**
current master 0fc06d657d596adcc289a5f228973ea268efd66d

**Additional context**
I will open a pr with fixes.

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
